Analysis/Tasks/PWGGA/gammaJet.cxx

```diff
@@ -0,0 +1,234 @@
+#include "Framework/ASoAHelpers.h"
+#include "ReconstructionDataFormats/Track.h"
+#include "AnalysisCore/RecoDecay.h"
+#include "AnalysisCore/trackUtilities.h"
+#include "AnalysisDataModel/StrangenessTables.h"
+#include "AnalysisCore/TrackSelection.h"
+#include "AnalysisDataModel/TrackSelectionTables.h"
+#include "AnalysisDataModel/EventSelection.h"
+#include "AnalysisDataModel/Centrality.h"
+
+#include "AnalysisDataModel/Jet.h"
+#include "AnalysisCore/JetFinder.h"
+#include "fastjet/PseudoJet.hh"
+#include "fastjet/ClusterSequenceArea.hh"
+
+#include <TH1F.h>
+#include <TMath.h>
+#include <TString.h>
+#include <iostream>
+#include <cmath>
+#include <TFile.h>
+#include <TH2F.h>
+#include <TProfile.h>
+#include <TLorentzVector.h>
+#include <Math/Vector4D.h>
+#include <TPDGCode.h>
+#include <TDatabasePDG.h>
+#include <array>
+#include <cstdlib>
+
+//using std::cout;
+//using std::endl;
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using std::array;
+
+
+struct ProcessGammaJet {
+
+  //general event histograms
+  OutputObj<TH1I> hist_events_N{TH1I("events_N", "events_N", 10, 0, 10)};
+  OutputObj<TH1F> hist_events_z{TH1F("events_z", "events_z", 100, -10., 10.)};
+  OutputObj<TH2F> hist_events_xy{TH2F("events_xy", "events_xy", 100, -1., 1., 100, -1., 1.)};
+  OutputObj<TH1I> hist_tracks_NGood{TH1I("tracks_NGood", "tracks_NGood", 1000, 0, 10000)};
+  OutputObj<TH2F> hist_tracks_etaphi{TH2F("tracks_etaphi", "tracks_etaphi", 102, -2.01, 2.01, 100, 0., 2. * M_PI)};
+
+  //photon histograms
+  OutputObj<TH1F> hist_calo_pt{TH1F("calo_pt", "calo_pt", 500, 0., 100.)};
+  OutputObj<TH1F> hist_calo_E{TH1F("calo_E", "calo_E", 500, 0., 100.)};
+  OutputObj<TH2F> hist_calo_etaphi{TH2F("calo_etaphi", "calo_etaphi", 102, -2.01, 2.01, 100, 0., 2. * M_PI)};
+
+  //jet histograms
+  OutputObj<TH1F> hist_jet_pt{TH1F("jet_pt", "jet_pt", 500, 0., 100.)};
+  OutputObj<TH2F> hist_jet_etaphi{TH2F("jet_etaphi", "jet_etaphi", 102, -2.01, 2.01, 100, 0., 2. * M_PI)};
+
+  //gamma-jet histograms
+  OutputObj<TH2F> hist_gammajet_detadphi{TH2F("gammajet_detadphi", "gammajet_detadphi", 102, -2.01, 2.01, 100, 0., 2. * M_PI)};
+  OutputObj<TH1F> hist_gammajet_dE{TH1F("gammajet_dE", "gammajet_dE", 100,0.,100.)};
+  OutputObj<TH1F> hist_gammajet_xj{TH1F("gammajet_xj", "gammajet_xj", 100,0.,10.)};
+  OutputObj<TH1F> hist_gammajet_ratioE{TH1F("gammajet_ratioE", "gammajet_ratioE", 100,0.,10.)};
+
+  //meson candidate histograms
+  OutputObj<TH2F> hist_meson_masspt{TH2F("meson_masspt", "meson_masspt", 100, 0., 1., 250, 0., 100.)};
+  
+
+  //the following doesn't work, why?
+  //hist_NEvents->GetXaxis()->SetBinLabel(0,"accepted");
+
+ //use a filter for tracks
+  Filter trackCuts = aod::track::pt >= 0.15f && aod::track::eta > -0.9f && aod::track::eta < 0.9f;
+
+  //initialize the jet finder
+  std::vector<fastjet::PseudoJet> jets;
+  std::vector<fastjet::PseudoJet> inputParticles;
+  JetFinder jetFinder;
```

Comment:
Should we use the jet finder task and subscribe to those tables here, instead of running it yourself?